services/lobby.service.js

```diff
@@ -0,0 +1,1003 @@
+            lobby._id.toString(),
+            { lobbyLocked: true },
+            { new: true }
+        );
+
+        // Start the round and get the updated lobby data
+        const result = await startRound(updatedLobby);
+        return {
+            code: 'success',
+            message: Constants.GAME_PLAYED_SUCCESSFULLY,
+            lobby: result,
+        };
+    } catch (error) {
+        console.log(error);
+        return Constants.INTERNAL_SEVER_ERROR;
+    }
+};
+
+
+
+const statusLobby = async (req, res) => {
+    try {
+        const { lobbyId, userId, status } = req.body;
+        const lobby = await Lobby.findOne({ _id: lobbyId });
+
+        if (!lobby) {
+            return 'Lobby not found.';
+        }
+
+        if (!status && lobby.playerList.includes(userId)) {
+            // Remove userId from the playerList using array filtering
+            lobby.playerList = lobby.playerList.filter((player) => player !== userId);
+
+            if (lobby.lobbyCreator.toString() === userId) {
+                lobby.lobbyCreator = lobby.playerList[0];
+            }
+
+            await lobby.save();
+        }
+
+        return lobby;
+
+    } catch (error) {
+        console.log(error, 'error')
+        return 'Internal server error.';
+    }
+};
+
+
+const updateRound = async (_id, round) => {
+    const result = await Lobby.findByIdAndUpdate(
+        _id,
+        { $push: { rounds: { round: round, questions: [] } } },
+        { new: true }
+    )
+    console.log(result, 'asdasdasdas')
+    return result
+}
+
+
+const startRound = async (req, res) => {
+    try {
+        const { body } = req;
+        if (body) {
+            console.log(body, 'body');
+            let findLobby = await Lobby.findOne({ lobbyCode: body.lobbyCode });
+            console.log('11111111111111111111');
+            if (body.round === '0') {
+                return 'Lobby match not start yet.'
+            }
+            else if (body.round === '1') {
+                if (findLobby?.rounds[0]?.loop1?.length > 0 && findLobby?.rounds[0]?.loop2?.length) {
+                    return "mafi de dao"
+                }
+                if (findLobby?.rounds[0]?.loop1?.length > 0) {
+
+                    console.log('222222222222222222');
+
+                    const questions = findLobby?.rounds[0]?.loop1;
+                    const users = findLobby.playerList;
+
+                    const loop2 = [];
+
+                    for (let outer = 0; outer < questions.length; outer++) {
+                        const round = [];
+
+                        for (let inner = 0; inner < users.length; inner++) {
+                            const targetQuestion = questions[(inner + outer) % questions.length];
+                            round.push({ lobbyUserId: users[inner], question: targetQuestion.question });
+
+                        }
+                        loop2.push(round);
+                    }
```

Comment:
Looks like this logic is being reused in multiple places. Can we abstract it away into a utility function?